static-site/src/components/CardsV2/helpers.js

```diff
@@ -0,0 +1,322 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * React hook which returns a sorted and filtered set of cards
+ * @param {string} method 
+ * @param {[]} selectedFilterOptions 
+ * @param {Card[]} cards 
+ * @param {function} setOrderedData 
+ * @returns {Card[]} same data structure as @param cards
+ */
+export const useSortAndFilter = (method, selectedFilterOptions, cards, setOrderedData) => {
+  useEffect(() => {
+    if (!cards) return;
+    console.log(`Sorting by ${method} ` + (selectedFilterOptions.length ? `filtering to ${selectedFilterOptions.map((el) => el.value).join(", ")}` : '(no filtering)'))
+
+    let data = clone(cards)
+    data = filterCards(data, selectedFilterOptions);
+
+    // Certain calculations (e.g. number of datasets) must be calculated after any filtering
+    // (and similarly, the pruning of cards only makes sense after filtering)
+    calculateCounts(data)
+    data = pruneCards(data)    
+    backpropagateInformation(data)
+
+    sortCards(data, method)
+
+    setOrderedData(data);
+
+  }, [cards, method, selectedFilterOptions, setOrderedData])
+
+} 
+
+/**
+ * React hook which scans the provided data (cards) and returns the
+ * available filtering options. These options are dependent on the
+ * current subset of cards displayed -- e.g. if you've previously
+ * filtered by "flu", then the updated set of available filtering
+ * options should only consider keywords in flu-like datasets.
+ */
+export function useFilterOptions(data) {
+  const [state, setState] = useState([]);
+  const counts = {};
+  useEffect(() => {
+    console.log("----- useEffect // useFilterOptions")
```

Comment:
Logs will be removed before merge, but they may be helpful for people during review